### DISCUSSION TITLE
How do I fix the timestamp in the startup log? (Linux x86_64 native)

### DISCUSSION THREAD

**Discussion by trholding** [nasa/cFS#405] [Q&A] [answered]
New here and getting used to cFE.

What I noticed is the timestamps shown here seem to be very off. How do I fix it? Any pointers welcome.

```
SCH Lab Initialized. SCH Lab DEVELOPMENT BUILD v2.5.0-rc4+dev6, Last Official Release: v2.3.0
1980-012-14:03:20.60514 CFE_ES_Main: CFE_ES_Main entering APPS_INIT state
1980-012-14:03:20.60517 CFE_ES_Main: CFE_ES_Main entering OPERATIONAL state
EVS Port1 66/1/CFE_TIME 21: Stop FLYWHEEL
```

**Comment by skliper** [comment, marked as answer]
Hello and welcome!  cFE has it's own internal timekeeping via the TIME service, aka "spacecraft" time.  It's worth noting that time services is fairly complex and was designed to handle a variety of use cases, I won't cover everything here but hopefully this will get you started.
The basics are there's the epoch that's set via configuration, a mission elapsed time (MET) counter, and a spacecraft time correlation factor (STCF).

The easiest way to update time is via the cFE set time command `CFE_TIME_SET_TIME_CC` or cFE set STCF command `SC_TIME_SET_STCF_CC`, both of which will adjust the spacecraft time correlation factor (STCF).  You could change the epoch, but that's a compile time configuration value.  You could also add a simple app that just grabs the time from the OS (or whatever real time clock is available) and sets the STCF for you.  More complex systems and/or those with tighter timing requirements typically involve a "tone" and "time at tone", and cFE can act as a timer server or client.

Note there are a few issues open to improve things, specifically nasa/cfe#908 to initialize the STCF from within the PSP, and more general nasa/cfe#302 to really simplify things.  Adding in a simple mechanism to update spacecraft time using OS time would be a good improvement.

**Reply by trholding** [reply]
Thanks for the great answer and great pointers. I am still wrapping my head around cFE. It will be challenging and rewarding. I hope to contribute as soon as possible. I am also developing a low cost Single Board Computer. I hope to get cFE to work on it, so that makers and students can use that to learn how to code for spacecraft.
